src/plots/linePlot.ts

```diff
@@ -22,6 +31,78 @@ export module Plots {
       this.attr("stroke-width", "2px");
     }
 
+    public x(): Plots.AccessorScaleBinding<X, number>;
+    public x(x: number | Accessor<number>): Line<X>;
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Line<X>;
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
```

Comment:
I find that we often need to do additional work in property setters, leading to overriding in the subclass with a duplication of superclass logic. Instead of duplicating the logic each time we have subclass-specific code in the setter, does it make sense to make a `protected` `_setX()` method that could be overridden independently?

This refactor shouldn't be done in this PR, of course.